saleor/payment/utils.py

```diff
@@ -125,9 +125,78 @@ def clean_authorize(payment: Payment):
         raise PaymentError('Charged transactions cannot be authorized again.')
 
 
+def call_gateway(
```

Comment:
That's the thing and the reason why something like `process_payment` exists, Saleor has no way of knowing what the gateway implements. Otherwise we could just call `auth`, then `capture` every time and what the gateway does.. we essentially don't care. But I feel that would be worse than making `process_payment` method, that we expect to actually process the payment (be it auth and .capture or charge). I don't really dig returning a list, but we might end up doing that in the end.

Reply:
Why not let the gateway itself returns a list of gateway response with KIND information. Then ```process_payment``` only need to iterate it and create different transactions.